probe/overlay/weave.go

```diff
@@ -30,14 +29,45 @@ type Weave struct {
 	url string
 }
 
+type weaveStatus struct {
+	Router struct {
+		Peers []struct {
+			Name     string `json:"name"`
+			NickName string `json:"nickname"`
+		} `json:"peers"`
+	} `json:"router"`
+}
+
 // NewWeave returns a new Weave tagger based on the Weave router at
 // address. The address should be an IP or FQDN, no port.
 func NewWeave(weaveRouterAddress string) (*Weave, error) {
-	s, err := sanitize("http://", 6784, "/status-json")(weaveRouterAddress)
+	s, err := sanitize("http://", 6784, "/report")(weaveRouterAddress)
 	if err != nil {
 		return nil, err
 	}
-	return &Weave{s}, nil
+	return &Weave{
+		url: s,
+	}, nil
+}
+
+func (w Weave) update() (weaveStatus, error) {
```

Comment:
I moved this out of Report() as I was reusing it in Tag(); but then I decided to save that for another change.